sdk/search/azure-search-documents/azure/search/documents/indexes/models/_models.py

```diff
@@ -3,24 +3,378 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
+from enum import Enum, EnumMeta
+from six import with_metaclass
 import msrest.serialization
 from .._generated.models import (
     LexicalAnalyzer,
     LexicalTokenizer,
     AnalyzeRequest,
     CustomAnalyzer as _CustomAnalyzer,
+    EntityRecognitionSkill as _EntityRecognitionSkillV1,
+    EntityRecognitionSkillV3 as _EntityRecognitionSkillV3,
     PatternAnalyzer as _PatternAnalyzer,
     PatternTokenizer as _PatternTokenizer,
     SearchResourceEncryptionKey as _SearchResourceEncryptionKey,
     SearchIndexerDataSource as _SearchIndexerDataSource,
+    SearchIndexerSkill,
+    SearchIndexerSkillset as _SearchIndexerSkillset,
+    SentimentSkill as _SentimentSkillV1,
+    SentimentSkillV3 as _SentimentSkillV3,
     SynonymMap as _SynonymMap,
     DataSourceCredentials,
     AzureActiveDirectoryApplicationCredentials,
 )
+from .._generated.models._search_client_enums import _CaseInsensitiveEnumMeta
```

Comment:
Maybe we should use azure.core.CaseInsensitiveEnumMeta

Reply:
If we did, we would be the only SDK doing so. It seems all SDKs declare this class in their `_foo_client_enum.py` file. Presumably, this is part of the autorest generation.

Reply:
The problem is if we reply on auto-generated private code, it may be subject to change(break). Of course, we can fix that when it breaks.

Reply:
As it turns out, using `CaseInsensitiveEnum` at all in custom code causes pylint to vomit in a way that you can't disable. Anna told me to make it a regular, case sensitive enum, which fixed the problem.

Reply:
I created an issue to track this known bug/limitation:
https://github.com/Azure/azure-sdk-for-python/issues/20520